.github/workflows/upload-test-stats-while-running.yml

```diff
@@ -0,0 +1,40 @@
+name: Upload test stats while running
+
+on:
+  schedule:
+    # Every hour
+    - cron: '0 * * * *'
```

Comment:
I wonder if this is the best way to run this script.  From what I see, the script will run continuously until it's cancelled by the next one.

* What do you think about having this upload script as a lambda that listen to S3 create event on `workflows_failing_pending_upload`?  This seems a better choice than a cron job
* If you think we can keep this setup for its simplicity, we need to tweak HUD a bit to hide the cancellation signals from this job

Reply:
Last I checked, lambdas do not have enough memory to run this

Reply:
I probably missed something as it's not clear where it memory bottleneck is there. This looks like a normal process of moving data from one place to another.  Is this the list of tests in https://github.com/pytorch/pytorch/blob/main/tools/stats/upload_test_stats.py#L116?

On the other hand, memory and the lambda timeout value should be configurable to a certain degree 
 https://docs.aws.amazon.com/lambda/latest/dg/configuration-memory.html

Reply:
I might be remembering something else then.  I do know it takes some amount of memory and time to unzip and parse all the test cases since I think I used to do something similar for the clickhouse s3 replicator.  My two concerns are that I'm not sure how I feel about duplicating this code across pytorch and either test-infra or pytorch-gha-infra to upload it to aws, and concurrency since the current code doesn't have any, but multiple lambdas could be running at the same time for the same workflow.  At the very least I think it would result in more invocations because you'd be processing every time something wants to upload to that bucket while this one batches the results by not processing immediately due to not running in parallel

Reply:
Yeah, the code duplication part is the ugly part